homeassistant/components/modbus/switch.py

```diff
@@ -111,32 +116,28 @@ async def async_turn_on(self, **kwargs):
         result = await self._hub.async_pymodbus_call(
             self._slave, self._address, self._command_on, self._write_type
         )
-        if result is False:
+        if result is None:
             self._available = False
-            self.async_write_ha_state()
         else:
-            self._available = True
             if self._verify_active:
                 await self.async_update()
             else:
                 self._is_on = True
-                self.async_write_ha_state()
+        self.async_write_ha_state()
```

Comment:
There may be double state updates with this change since we update state in `async_update` too. We should avoid that.

Reply:
I knew it, being lazy does not pay off 😄 Corrected for both turn_on/turn_off.

I made a change for KapJI while on the move, and it seems "black" did not like it, but let us see.

Another question: You always change the title to the same, I assume it for the automations on the project dev page ?

Reply:
It's not black didn't like it, it's Github Actions are not working 🙂
https://www.githubstatus.com/

The new title is without dot, it's usually not used in the title.

Reply:
I knew it !! we need our bot to inform us in discord when GitHub have problems.

Thanks now I know why he keeps changing title, I am on a couple of other projects and they are keen on the ".", as they otherwise it is open ended 😄

Reply:
Yes, we prefer a title without period at the end and preferably also written in imperative mode. Eg: Fix modbus result handling

Reply:
Already changed in the other PR 😄